pkg/api/handlers/compat/images_build.go

```diff
@@ -399,19 +399,38 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// Docker's newer clients popuates `cacheFrom` and `cacheTo` parameter
+	// by default as empty array for all commands but buildah's design of
+	// distributed cache expects this to be a repo not image hence parse
+	// only the first populated repo and igore if empty array.
+	// Read more here: https://github.com/containers/podman/issues/15928
+	compatIgnoreForcedCacheOptions := func(queryStr string) string {
+		query := queryStr
+		if strings.Contains(query, "[") {
+			var arr []string
+			parseErr := json.Unmarshal([]byte(query), &arr)
+			if parseErr != nil {
+				query = arr[0]
```

Comment:
this will panic with index out of bounds for an empty slice.
Also if docker allows an array and we are not then, this is not compatible with docker.

Reply:
We can implement accepting multiple repos which should make it API compatible but not functionally same but that is a different feature at buildah end.